lib/Default/AbstractSmrShip.class.inc

```diff
@@ -507,7 +506,7 @@ abstract class AbstractSmrShip {
 		if($this->getHardware($hardwareTypeID) == $amount)
 			return;
 		$this->hardware[$hardwareTypeID] = $amount;
-		$this->hasChangedHardware = true;
+		$this->hasChangedHardware[] = $hardwareTypeID;
```

Comment:
It would be better to do
```php
$this->hasChangedHardware[$hardwareTypeID] = true
```
that way you guarantee the `$hardwareTypeID` only appears once, with the current method doing
```php
$ship->setHardware(HARDWARE_SHIELDS, 15)
$ship->setHardware(HARDWARE_SHIELDS, 10)
$ship->setHardware(HARDWARE_SHIELDS, 5)
```
would result in 3 database calls at the end (and I wouldn't be surprised at all if combat triggers a `setHardware` for every weapon hit)

Reply:
Good point. I actually had that to begin with, but then I changed it because I didn't want to do the extra check that the value was `true` (even though we probably don't need to do it). I'll change it back. Thanks!

Reply:
Okay, this change has been made.